key-export-common/src/lib.rs

```diff
@@ -0,0 +1,198 @@
+#[serde_as]
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct EncryptedShareAndIdentity {
+    /// Signer's identity
+    #[serde_as(as = "Base64")]
+    pub signer_identity: Vec<u8>,
+    /// Signers's key share.
+    /// It is an encrypted `dfns_key_export_common::KeySharePlaintext`.
+    /// Ciphertext and plaintext are in format defined by `dfns-key-import-common`
+    /// library. See https://github.com/dfns-labs/trusted-dealer/
+    #[serde_as(as = "Base64")]
+    pub encrypted_key_share: Vec<u8>,
+}
+
+/// Key export request that's intended to be sent fromt the client
+/// to Dfns API.
+#[serde_as]
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct KeyExportRequest {
+    /// The wallet-id whose private key will be extracted
+    pub wallet_id: String,
+    /// An encryption key, to be used by signers to sign their key share.
+    ///
+    /// It contains the bytes of a `dfns_key_import_common::encryption::EncryptionKey`,
+    /// as defined in `dfns-key-import-common` library:
+    /// https://github.com/dfns-labs/trusted-dealer/
+    #[serde_as(as = "Base64")]
+    pub encryption_key: Vec<u8>,
+    /// Key types (protocol and curve) supported by the WASM module
+    /// that generated the KeyExportRequest.
+    pub supported_schemes: Vec<SupportedScheme>,
+}
+
+/// Key export response, sent from Dfns API to the WASM module.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct KeyExportResponse {
+    /// The threshold of the specified wallet.
+    pub min_signers: u32,
+    /// The public key of the specified wallet.
+    #[serde(with = "hex::serde")]
+    pub public_key: Vec<u8>,
+    /// The protocol the exported scheme can be used for
+    pub protocol: KeyProtocol,
+    /// The curve the exported scheme can be used for
+    pub curve: KeyCurve,
+    /// Identities and encrypted shares of wallet's key holders.
+    pub encrypted_shares: Vec<EncryptedShareAndIdentity>,
+}
+
+/// The protocol and curve for which a key can be used
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub struct SupportedScheme {
+    /// protocol
+    pub protocol: KeyProtocol,
+    /// curve
+    pub curve: KeyCurve,
+}
+
+/// The protocol for which a key can be used
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub enum KeyProtocol {
+    ///GG18
+    Gg18,
+    ///Binance EDDSA
+    BinanceEcdsa,
+    ///CGGMP21
+    Cggmp21,
+}
+
+/// The curve for which a key can be used
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+pub enum KeyCurve {
+    /// Secp256k1 curve
+    Secp256k1,
+    /// Ed25519 curve
+    Ed25519,
+}
+
+/// Internal function to perform interpolation.
+/// In the end tt verifies the computed key against the provided
+/// public key and returns an error if it doesn't match.
+///
+/// `key_shares` is a vector of serialized KeySharePlaintext<E>
+/// 'public_key` is a serialized Point<E>
+pub fn interpolate_secret_key<E: Curve>(
```

Comment:
Missing empty line between header and main body of the doc comment

I advice to do `cargo doc --open` to check that all documentation is displayed as intended

```suggestion
/// Internal function to perform interpolation.
///
/// In the end tt verifies the computed key against the provided
/// public key and returns an error if it doesn't match.
///
/// `key_shares` is a vector of serialized KeySharePlaintext<E>
/// 'public_key` is a serialized Point<E>
pub fn interpolate_secret_key<E: Curve>(
```